### PR TITLE
Re order

### PR DESCRIPTION
**Pull request by jburel**
Re-organise notebook as discussed
cc @will-moore @pwalczysko

### CODE REVIEW

File: notebooks/idr0062_prediction.ipynb [hunk 1]
```diff
@@ -476,14 +467,76 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "4d4fac89",
+   "metadata": {},
+   "source": [
+    "## Save the StarDist labels\n",
+    "\n",
+    "Stardist offers method to save the labels into ImageJ rois using ``export_imagej_rois``. This is outside the scope of this notebook. \n",
+    "\n",
+    "Below we show how to ave the labels locally in a machine and human readable format: **geojson**.\n",
```
Comment (will-moore) [inline notebooks/idr0062_prediction.ipynb]: ave

**Comment by pwalczysko** [comment]
In Google Colab, I am getting
Screenshot 2022-04-25 at 14 52 18

Edit: But executed all cells successfully in Google Colab in the end.

**Comment by pwalczysko** [comment]
MyBinder and  seems to have a problem

on cell 
**Load stardist trained model**

Possibly this is because the kernel selection is wrong ? I cannot select ``stardist`` as kernel, because it is not offered. If I select the ``idr0062... preferred session`` option, it does not work.

Screenshot 2022-04-25 at 15 22 45


<details>

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [10], in <module>
      1 from stardist.models import StarDist2D
----> 2 model_versatile = StarDist2D.from_pretrained('2D_demo')

File /srv/conda/envs/notebook/lib/python3.9/site-packages/csbdeep/models/base_model.py:79, in BaseModel.from_pretrained(cls, name_or_alias)
     77 try:
     78     get_model_details(cls, name_or_alias, verbose=True)
---> 79     return get_model_instance(cls, name_or_alias)
     80 except ValueError:
     81     if name_or_alias is not None:

File /srv/conda/envs/notebook/lib/python3.9/site-packages/csbdeep/models/pretrained.py:102, in get_model_instance(cls, key_or_alias)
    100 def get_model_instance(cls, key_or_alias):
    101     path = get_model_folder(cls, key_or_alias)
--> 102     model = cls(config=None, name=path.stem, basedir=path.parent)
    103     model.basedir = None # make read-only
    104     return model

File /srv/conda/envs/notebook/lib/python3.9/site-packages/stardist/models/model2d.py:292, in StarDist2D.__init__(self, config, name, basedir)
    290 def __init__(self, config=Config2D(), name=None, basedir='.'):
    291     """See class docstring."""
--> 292     super().__init__(config, name=name, basedir=basedir)

File /srv/conda/envs/notebook/lib/python3.9/site-packages/stardist/models/base.py:222, in StarDistBase.__init__(self, config, name, basedir)
    221 def __init__(self, config, name=None, basedir='.'):
--> 222     super().__init__(config=config, name=name, basedir=basedir)
    223     threshs = dict(prob=None, nms=None)
    224     if basedir is not None:

File /srv/conda/envs/notebook/lib/python3.9/site-packages/csbdeep/models/base_model.py:113, in BaseModel.__init__(self, config, name, basedir)
    111 self.keras_model = self._build()
    112 if config is None:
--> 113     self._find_and_load_weights()

File /srv/conda/envs/notebook/lib/python3.9/site-packages/csbdeep/models/base_model.py:32, in suppress_without_basedir.<locals>._suppress_without_basedir.<locals>.wrapper(*args, **kwargs)
     30     warn is False or warnings.warn("Suppressing call of '%s' (due to basedir=None)." % f.__name__)
     31 else:
---> 32     return f(*args, **kwargs)

File /srv/conda/envs/notebook/lib/python3.9/site-packages/csbdeep/models/base_model.py:167, in BaseModel._find_and_load_weights(self, prefer)
    165 weights_chosen = weights_preferred[0] if len(weights_preferred)>0 else weights_files[0]
    166 print("Loading network weights from '%s'." % weights_chosen.name)
--> 167 self.load_weights(weights_chosen.name)

File /srv/conda/envs/notebook/lib/python3.9/site-packages/csbdeep/models/base_model.py:32, in suppress_without_basedir.<locals>._suppress_without_basedir.<locals>.wrapper(*args, **kwargs)
     30     warn is False or warnings.warn("Suppressing call of '%s' (due to basedir=None)." % f.__name__)
     31 else:
---> 32     return f(*args, **kwargs)

File /srv/conda/envs/notebook/lib/python3.9/site-packages/csbdeep/models/base_model.py:184, in BaseModel.load_weights(self, name)
    175 @suppress_without_basedir(warn=True)
    176 def load_weights(self, name='weights_best.h5'):
    177     """Load neural network weights from model folder.
    178 
    179     Parameters
   (...)
    182         Name of HDF5 weight file (as saved during or after training).
    183     """
--> 184     self.keras_model.load_weights(str(self.logdir/name))

File /srv/conda/envs/notebook/lib/python3.9/site-packages/tensorflow/python/keras/engine/training.py:2234, in Model.load_weights(self, filepath, by_name, skip_mismatch, options)
   2231   hdf5_format.load_weights_from_hdf5_group_by_name(
   2232       f, self.layers, skip_mismatch=skip_mismatch)
   2233 else:
-> 2234   hdf5_format.load_weights_from_hdf5_group(f, self.layers)

File /srv/conda/envs/notebook/lib/python3.9/site-packages/tensorflow/python/keras/saving/hdf5_format.py:662, in load_weights_from_hdf5_group(f, layers)
    651 """Implements topological (order-based) weight loading.
    652 
    653 Arguments:
   (...)
    659         and weights file.
    660 """
    661 if 'keras_version' in f.attrs:
--> 662   original_keras_version = f.attrs['keras_version'].decode('utf8')
    663 else:
    664   original_keras_version = '1'

AttributeError: 'str' object has no attribute 'decode'
```

</details>

**Comment by pwalczysko** [comment]
the same problem and error as on MyBinder is on repo2docker. Also the same situation with the kernel there.

**Comment by pwalczysko** [comment]
On conda, all works fine. I also like the order of cells.

**Comment by pwalczysko** [comment]
In summary

- Google Colab and conda -> as expected
- MyBinder and repo2docker -> error as highlighted in https://github.com/ome/omero-guide-python/pull/49#issuecomment-1108631138

File: notebooks/idr0062_prediction.ipynb [hunk 2]
```diff
@@ -344,65 +320,80 @@
...
    "cell_type": "markdown",
-   "id": "78d8d0dc",
+   "id": "5ddfa3f8",
    "metadata": {},
    "source": [
-    "## Save the Stardist labels\n",
-    "Save the labels locally in a machine and human readable format: geojson.\n",
-    "Stardist offers method to save the labels into ImageJ rois using ``export_imagej_rois``. This is outside the scope of this notebook.\n"
+    "## Load the original labels\n",
+    "Load the original labels so we can compare them with the Stardist ones\n",
```
Comment (pwalczysko) [inline notebooks/idr0062_prediction.ipynb]:
```suggestion
    "Load the original labels in order to compare them with the Stardist ones\n",
```

small typo

**Comment by jburel** [comment]
@pwalczysko it should be sorted now

**Comment by pwalczysko** [comment]
No luck on MyBinder, same experience. Tried a new browser, did not help

**Comment by pwalczysko** [comment]
might have been a caching issue. It built it very very quickly. I am not using the badge now, but rather rebuilding it from MyBinder start page. Takes much longer.

Edit: sorry, not true. I was just putting in wrong params. With the right ones, it takes very shortly.

**Comment by pwalczysko** [comment]
Indeed, whatever I do, MyBinder fails with the same error as before. And no possiblity to choose stardist kernel.

**Comment by pwalczysko** [comment]
And repo2docker fails too, exactly as before.

File: notebooks/idr0062_prediction.ipynb [hunk 3]
```diff
@@ -476,14 +467,76 @@
...
+   "id": "4d4fac89",
+   "metadata": {},
+   "source": [
+    "## Save the StarDist labels\n",
+    "\n",
+    "Stardist offers method to save the labels into ImageJ rois using ``export_imagej_rois``. This is outside the scope of this notebook. \n",
+    "\n",
+    "Below we show how to save the labels locally in a machine and human readable format: **geojson**.\n",
+    "\n",
+    "* Convert the Stardist labels into geojson Polygons\n",
```
Comment (pwalczysko) [inline notebooks/idr0062_prediction.ipynb]:
```suggestion
    "* Convert the Stardist polygon coordinates into geojson Polygons\n",
```

I think you are not using the label image here, just the ``results_details`` which are described as polygon details above.

File: notebooks/idr0062_prediction.ipynb [hunk 4]
```diff
@@ -476,14 +467,76 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "4d4fac89",
+   "metadata": {},
+   "source": [
+    "## Save the StarDist labels\n",
+    "\n",
+    "Stardist offers method to save the labels into ImageJ rois using ``export_imagej_rois``. This is outside the scope of this notebook. \n",
+    "\n",
+    "Below we show how to save the labels locally in a machine and human readable format: **geojson**.\n",
```
Comment (pwalczysko) [inline notebooks/idr0062_prediction.ipynb]:
```suggestion
    "Below we show how to save the segmentation represented as polygon details locally in a machine- and human-readable format: **geojson**.\n",
```

**Comment by pwalczysko** [comment]
Unfortunately, ``repo2docker`` still fails with the same error as before. Definitely checked out the last commit and rerun the ``repo2docker .`` command.

```
commit 2345b962a7cc0d12b33505c44391fc2b28dd9fa4 (HEAD, jburel/re-order)
Author: Jean-Marie Burel <j.burel@dundee.ac.uk>
Date:   Wed Apr 27 12:27:39 2022 +0100

    add keras if using colab
...
```

**Comment by pwalczysko** [comment]
The same for MyBinder. No change in behaviour.

The error is as before, long output saying

```
...
AttributeError: 'str' object has no attribute 'decode'
```

**Comment by pwalczysko** [comment]
1. MyBinder passing
2. repo2docker passing

testing Colab and conda now

**Comment by pwalczysko** [comment]
3. Colab works
4. conda works

**Comment by pwalczysko** [comment]
Thank you. Ready to merge.

**Review by pwalczysko — APPROVED** [review]

**Pull request merged**